Jazzy+ tests fail

It seems that there is probably some timing issues in the tests, easy to reproduce locally.

For basic_replay

```
[INFO] [launch]: All log files can be found below /home/emerson/.ros/log/2025-05-12-14-00-03-336192-7832d40c1cca-3349                                                                                                                                                                                            
[INFO] [launch]: Default logging verbosity is set to INFO                                                                                                                                                                                                                                                        
[INFO] [ros2-1]: process started with pid [3367]                                                                                                                                                                                                                                                                 
[INFO] [topic_pub-2]: process started with pid [3368]                                                                                                                                                                                                                                                            
[INFO] [ros2_bag_player-3]: process started with pid [3369]
[ros2_bag_player-3] stdin is not a terminal device. Keyboard handling disabled.[INFO] [1747083603.507150132] [rosbag2_player]: Set rate to 1                                                                                                                                                                     
[ros2_bag_player-3] [INFO] [1747083603.509559069] [rosbag2_player]: Adding keyboard callbacks.                                                                                                                                                                                                                   
[ros2_bag_player-3] [INFO] [1747083603.509569651] [rosbag2_player]: Press SPACE for Pause/Resume                                                                                                                                                                                                                 
[ros2_bag_player-3] [INFO] [1747083603.509572047] [rosbag2_player]: Press CURSOR_RIGHT for Play Next Message                                                                                                                                                                                                     
[ros2_bag_player-3] [INFO] [1747083603.509574097] [rosbag2_player]: Press CURSOR_UP for Increase Rate 10%                                                                                                                                                                                                        
[ros2_bag_player-3] [INFO] [1747083603.509575784] [rosbag2_player]: Press CURSOR_DOWN for Decrease Rate 10%                                                                                                                                                                                                      
[ros2_bag_player-3] [INFO] [1747083603.509628580] [rosbag2_player]: Playback until timestamp: -1                                                                                                                                                                                                                 
[ros2-1] stdin is not a terminal device. Keyboard handling disabled.[INFO] [1747083603.511918385] [rosbag2_recorder]: Press SPACE for pausing/resuming                                                                                                                                                           
[ros2-1] [INFO] [1747083603.513332630] [rosbag2_recorder]: Listening for topics...                                                                                                                                                                                                                               
[ros2-1] [INFO] [1747083603.513345634] [rosbag2_recorder]: Event publisher thread: Starting                                                                                                                                                                                                                      
[ros2-1] [INFO] [1747083603.513827776] [rosbag2_recorder]: Subscribed to topic '/rosout'                                                                                                                                                                                                                         
[ros2-1] [INFO] [1747083603.514553437] [rosbag2_recorder]: Subscribed to topic '/vehicle/cmd_vel'                                                                                                                                                                                                                
[ros2-1] [INFO] [1747083603.514817968] [rosbag2_recorder]: Subscribed to topic '/parameter_events'                                                                                                                                                                                                               
[ros2-1] [INFO] [1747083603.515041430] [rosbag2_recorder]: Subscribed to topic '/events/write_split'                                                                                                                                                                                                             
[ros2-1] [INFO] [1747083603.515276825] [rosbag2_recorder]: Subscribed to topic '/events/read_split'                                                                                                                                                                                                              
[ros2-1] [INFO] [1747083603.515594608] [rosbag2_recorder]: Subscribed to topic '/clock'                                                                                                                                                                                                                          
[ros2-1] [INFO] [1747083603.515631765] [rosbag2_recorder]: Recording...
[topic_pub-2] publisher: beginning loop
[topic_pub-2] publishing #1: geometry_msgs.msg.Twist(linear=geometry_msgs.msg.Vector3(x=1.0, y=0.0, z=0.0), angular=geometry_msgs.msg.Vector3(x=0.0, y=0.0, z=0.5))
[topic_pub-2]                                                               
[ros2-1] [INFO] [1747083604.119294488] [rosbag2_recorder]: Subscribed to topic '/user/cmd_vel'                                                                                                                                                                                                                   
[INFO] [ros2_bag_player-3]: process has finished cleanly [pid 3369]
[INFO] [topic_pub-2]: sending signal 'SIGINT' to process[topic_pub-2]
[INFO] [ros2-1]: sending signal 'SIGINT' to process[ros2-1]
[ros2-1] [INFO] [1747083604.621510486] [rosbag2_recorder]: Pausing recording.                                                                                                                                                                                                                                    
[ros2-1] [INFO] [1747083604.621663805] [rosbag2_cpp]: Writing remaining messages from cache to the bag. It may take a while                                                                                                                                                                                      
[ros2-1] [INFO] [1747083604.630357635] [rosbag2_recorder]: Event publisher thread: Exiting                                                                                                                                                                                                                       
[ros2-1] [INFO] [1747083604.630387578] [rosbag2_recorder]: Recording stopped                                                                                                                                                                                                                                     
[ros2-1] [INFO] [1747083604.631818433] [rclcpp]: signal_handler(signum=2)
[ERROR] [topic_pub-2]: process has died [pid 3368, exit code 2, cmd 'ros2 topic pub /user/cmd_vel geometry_msgs/msg/Twist {linear: {x: 1.0}, angular: {z: 0.5}}'].
[INFO] [ros2-1]: process has finished cleanly [pid 3367]
============================================================
=================== STAGE RUN COMPLETED ====================
============================================================
============================================================
================== STAGE ANALYZE STARTING ==================
============================================================
test_cmd_vel (replay_testing.replay_runner.ReplayTestingRunner.analyze.<locals>.AnalyzeWithReader.test_cmd_vel) ... FAIL                                                                                                                                                                                         

======================================================================
FAIL: test_cmd_vel (replay_testing.replay_runner.ReplayTestingRunner.analyze.<locals>.AnalyzeWithReader.test_cmd_vel)                                                                                                                                                                                            
----------------------------------------------------------------------
Traceback (most recent call last):                                          
  File "/ws/src/replay_testing/test/replay_tests/basic_replay.py", line 63, in test_cmd_vel                                                                                                                                                                                                                      
    assert len(msgs) == 1                                                   
           ^^^^^^^^^^^^^^                                                   
AssertionError                                                              

```


It looks like the order of operations is:
1. bag player starts
2. bag recorder starts
3. ros2 topic pub starts and publishes 1 message
4. bag recorder sees the new ros2 topic pub topic, creates subscription
5. player finishes playing
6. shutdown

Therefore the recorder never got to see a message from the `ros2 topic pub` CLI publisher. Not actually sure why this works in Humble. Maybe it's a Cyclone thing, maybe the CLI created the publisher at a longer delay, maybe it's a rosbag2 discovery thing. Regardless, the above order of events all seems kinda sensible except the "player finishes playing & shutdown" thing. Can't really pick apart what's being tested, the bag playback seems to just be in there as a way to "kill" time, then the test is just running rosbag2 record against a ros2 topic pub, and expecting messages to be there.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -28,6 +28,7 @@ jobs:
     env:
       ROS_DISTRO: ${{ matrix.ros }}
       PIP_BREAK_SYSTEM_PACKAGES: 1
+      USE_MCAP_PIP: 1
     steps:
       - uses: actions/checkout@v4
       - uses: ros-tooling/action-ros-ci@v0.4

--- a/README.md
+++ b/README.md
@@ -128,7 +128,7 @@ class Analyze:
         )
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) == 1
+        assert len(msgs) >= 1
         assert msgs[0].channel.topic == "/user/cmd_vel"
 ```
 
@@ -164,6 +164,8 @@ class Run:
                         "ros2",
                         "topic",
                         "pub",
+                        "-r",
+                        "10",
                         "/user/cmd_vel",
                         "geometry_msgs/msg/Twist",
                         "{linear: {x: 1.0}, angular: {z: 0.5}}",
@@ -183,7 +185,7 @@ class AnalyzeBasicReplay:
         )
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) == 1
+        assert len(msgs) >= 1
         assert msgs[0].channel.topic == "/user/cmd_vel"
 
 ```

--- a/package.xml
+++ b/package.xml
@@ -28,7 +28,7 @@
   <test_depend>python3-pytest</test_depend>
   <test_depend>ros2run</test_depend>
   <test_depend>ros2topic</test_depend>
-  <test_depend>mcap-ros2-support</test_depend>
+  <test_depend condition="$USE_MCAP_PIP == 1">mcap-ros2-support</test_depend>
 
   <export>
     <build_type>ament_cmake</build_type>

--- a/package.xml
+++ b/package.xml
@@ -12,7 +12,6 @@
   <buildtool_depend>ament_cmake</buildtool_depend>
 
   <exec_depend>launch</exec_depend>
-  <exec_depend>mcap-ros2-support</exec_depend>
   <exec_depend>python3-pydantic</exec_depend>
   <exec_depend>python3-termcolor</exec_depend>
   <!-- we never mention rclpy directly, but rosbag2_py in Humble doesn't specify its dependency properly -->
@@ -29,6 +28,7 @@
   <test_depend>python3-pytest</test_depend>
   <test_depend>ros2run</test_depend>
   <test_depend>ros2topic</test_depend>
+  <test_depend>mcap-ros2-support</test_depend>
 
   <export>
     <build_type>ament_cmake</build_type>

--- a/test/replay_tests/basic_replay.py
+++ b/test/replay_tests/basic_replay.py
@@ -26,15 +26,13 @@ from replay_testing import (
     run,
 )
 
-cmd_vel_only_fixture = (
-    pathlib.Path(__file__).parent.parent / "fixtures" / "cmd_vel_only.mcap"
-).as_posix()
+cmd_vel_only_fixture = (pathlib.Path(__file__).parent.parent / 'fixtures' / 'cmd_vel_only.mcap').as_posix()
 
 
 @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
 class Fixtures:
-    required_input_topics = ["/vehicle/cmd_vel"]
-    expected_output_topics = ["/user/cmd_vel"]
+    required_input_topics = ['/vehicle/cmd_vel']
+    expected_output_topics = ['/user/cmd_vel']
 
 
 @run.default()
@@ -43,17 +41,17 @@ class Run:
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    "ros2",
-                    "topic",
-                    "pub",
-                    "-r",
-                    "10",
-                    "/user/cmd_vel",
-                    "geometry_msgs/msg/Twist",
-                    "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                    'ros2',
+                    'topic',
+                    'pub',
+                    '-r',
+                    '10',
+                    '/user/cmd_vel',
+                    'geometry_msgs/msg/Twist',
+                    '{linear: {x: 1.0}, angular: {z: 0.5}}',
                 ],
-                name="topic_pub",
-                output="screen",
+                name='topic_pub',
+                output='screen',
             )
         ])
 
@@ -61,8 +59,8 @@ class Run:
 @analyze
 class AnalyzeBasicReplay:
     def test_cmd_vel(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
         msgs = [msg for msg in msgs_it]
         assert len(msgs) >= 1
-        assert msgs[0].channel.topic == "/user/cmd_vel"
+        assert msgs[0].channel.topic == '/user/cmd_vel'

--- a/test/replay_tests/basic_replay.py
+++ b/test/replay_tests/basic_replay.py
@@ -26,13 +26,15 @@ from replay_testing import (
     run,
 )
 
-cmd_vel_only_fixture = (pathlib.Path(__file__).parent.parent / 'fixtures' / 'cmd_vel_only.mcap').as_posix()
+cmd_vel_only_fixture = (
+    pathlib.Path(__file__).parent.parent / "fixtures" / "cmd_vel_only.mcap"
+).as_posix()
 
 
 @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
 class Fixtures:
-    required_input_topics = ['/vehicle/cmd_vel']
-    expected_output_topics = ['/user/cmd_vel']
+    required_input_topics = ["/vehicle/cmd_vel"]
+    expected_output_topics = ["/user/cmd_vel"]
 
 
 @run.default()
@@ -41,17 +43,17 @@ class Run:
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    'ros2',
-                    'topic',
-                    'pub',
-                    '-r',
-                    '10',
-                    '/user/cmd_vel',
-                    'geometry_msgs/msg/Twist',
-                    '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                    "ros2",
+                    "topic",
+                    "pub",
+                    "-r",
+                    "10",
+                    "/user/cmd_vel",
+                    "geometry_msgs/msg/Twist",
+                    "{linear: {x: 1.0}, angular: {z: 0.5}}",
                 ],
-                name='topic_pub',
-                output='screen',
+                name="topic_pub",
+                output="screen",
             )
         ])
 
@@ -59,8 +61,8 @@ class Run:
 @analyze
 class AnalyzeBasicReplay:
     def test_cmd_vel(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) > 1
-        assert msgs[0].channel.topic == '/user/cmd_vel'
+        assert len(msgs) >= 1
+        assert msgs[0].channel.topic == "/user/cmd_vel"

--- a/test/replay_tests/basic_replay.py
+++ b/test/replay_tests/basic_replay.py
@@ -44,6 +44,8 @@ class Run:
                     'ros2',
                     'topic',
                     'pub',
+                    '-r',
+                    '10',
                     '/user/cmd_vel',
                     'geometry_msgs/msg/Twist',
                     '{linear: {x: 1.0}, angular: {z: 0.5}}',
@@ -60,5 +62,5 @@ class AnalyzeBasicReplay:
         msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) == 1
+        assert len(msgs) > 1
         assert msgs[0].channel.topic == '/user/cmd_vel'

--- a/test/replay_tests/basic_replay_failure.py
+++ b/test/replay_tests/basic_replay_failure.py
@@ -26,15 +26,13 @@ from replay_testing import (
     run,
 )
 
-cmd_vel_only_fixture = (
-    pathlib.Path(__file__).parent.parent / "fixtures" / "cmd_vel_only.mcap"
-).as_posix()
+cmd_vel_only_fixture = (pathlib.Path(__file__).parent.parent / 'fixtures' / 'cmd_vel_only.mcap').as_posix()
 
 
 @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
 class FilterFixtures:
-    required_input_topics = ["/vehicle/cmd_vel"]
-    expected_output_topics = ["/user/cmd_vel"]
+    required_input_topics = ['/vehicle/cmd_vel']
+    expected_output_topics = ['/user/cmd_vel']
 
 
 @run.default()
@@ -43,17 +41,17 @@ class Run:
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    "ros2",
-                    "topic",
-                    "pub",
-                    "-r",
-                    "10",
-                    "/user/cmd_vel",
-                    "geometry_msgs/msg/Twist",
-                    "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                    'ros2',
+                    'topic',
+                    'pub',
+                    '-r',
+                    '10',
+                    '/user/cmd_vel',
+                    'geometry_msgs/msg/Twist',
+                    '{linear: {x: 1.0}, angular: {z: 0.5}}',
                 ],
-                name="topic_pub",
-                output="screen",
+                name='topic_pub',
+                output='screen',
             )
         ])
 
@@ -61,8 +59,8 @@ class Run:
 @analyze
 class AnalyzeBasicReplay:
     def test_expected_failure(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
         msgs = [msg for msg in msgs_it]
         assert len(msgs) >= 1
-        assert msgs[0].channel.topic == "/nonexistent/cmd_vel"
+        assert msgs[0].channel.topic == '/nonexistent/cmd_vel'

--- a/test/replay_tests/basic_replay_failure.py
+++ b/test/replay_tests/basic_replay_failure.py
@@ -26,13 +26,15 @@ from replay_testing import (
     run,
 )
 
-cmd_vel_only_fixture = (pathlib.Path(__file__).parent.parent / 'fixtures' / 'cmd_vel_only.mcap').as_posix()
+cmd_vel_only_fixture = (
+    pathlib.Path(__file__).parent.parent / "fixtures" / "cmd_vel_only.mcap"
+).as_posix()
 
 
 @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
 class FilterFixtures:
-    required_input_topics = ['/vehicle/cmd_vel']
-    expected_output_topics = ['/user/cmd_vel']
+    required_input_topics = ["/vehicle/cmd_vel"]
+    expected_output_topics = ["/user/cmd_vel"]
 
 
 @run.default()
@@ -41,15 +43,17 @@ class Run:
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    'ros2',
-                    'topic',
-                    'pub',
-                    '/user/cmd_vel',
-                    'geometry_msgs/msg/Twist',
-                    '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                    "ros2",
+                    "topic",
+                    "pub",
+                    "-r",
+                    "10",
+                    "/user/cmd_vel",
+                    "geometry_msgs/msg/Twist",
+                    "{linear: {x: 1.0}, angular: {z: 0.5}}",
                 ],
-                name='topic_pub',
-                output='screen',
+                name="topic_pub",
+                output="screen",
             )
         ])
 
@@ -57,8 +61,8 @@ class Run:
 @analyze
 class AnalyzeBasicReplay:
     def test_expected_failure(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) == 1
-        assert msgs[0].channel.topic == '/nonexistent/cmd_vel'
+        assert len(msgs) >= 1
+        assert msgs[0].channel.topic == "/nonexistent/cmd_vel"

--- a/test/replay_tests/basic_replay_multiple_fixtures_and_params.py
+++ b/test/replay_tests/basic_replay_multiple_fixtures_and_params.py
@@ -29,10 +29,10 @@ from replay_testing import (
     run,
 )
 
-replay_testing_dir = get_package_share_directory('replay_testing')
+replay_testing_dir = get_package_share_directory("replay_testing")
 
-cmd_vel_only_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only.mcap')
-cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only_2.mcap')
+cmd_vel_only_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only.mcap")
+cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only_2.mcap")
 
 
 @fixtures.parameterize([
@@ -40,33 +40,35 @@ cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'c
     McapFixture(path=cmd_vel_only_2_fixture),
 ])
 class Fixtures:
-    required_input_topics = ['/vehicle/cmd_vel']
-    expected_output_topics = ['/user/cmd_vel']
+    required_input_topics = ["/vehicle/cmd_vel"]
+    expected_output_topics = ["/user/cmd_vel"]
 
 
 @run.parameterize([
-    ReplayRunParams(name='run_1_twist_slow', params={'x': 1.0}),
-    ReplayRunParams(name='run_2_twist_fast', params={'x': 10.0}),
+    ReplayRunParams(name="run_1_twist_slow", params={"x": 1.0}),
+    ReplayRunParams(name="run_2_twist_fast", params={"x": 10.0}),
 ])
 class Run:
     def generate_launch_description(self, replay_run_params: ReplayRunParams) -> LaunchDescription:
-        print('replay_run_parms')
+        print("replay_run_parms")
         twist_msg = {
-            'linear': {'x': replay_run_params.params['x']},
-            'angular': {'z': 0.5},
+            "linear": {"x": replay_run_params.params["x"]},
+            "angular": {"z": 0.5},
         }
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    'ros2',
-                    'topic',
-                    'pub',
-                    '/user/cmd_vel',
-                    'geometry_msgs/msg/Twist',
+                    "ros2",
+                    "topic",
+                    "pub",
+                    "-r",
+                    "10",
+                    "/user/cmd_vel",
+                    "geometry_msgs/msg/Twist",
                     json.dumps(twist_msg),
                 ],
-                name='topic_pub',
-                output='screen',
+                name="topic_pub",
+                output="screen",
             )
         ])
 
@@ -74,8 +76,8 @@ class Run:
 @analyze
 class Analyze:
     def test_cmd_vel(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) == 1
-        assert msgs[0].channel.topic == '/user/cmd_vel'
+        assert len(msgs) >= 1
+        assert msgs[0].channel.topic == "/user/cmd_vel"

--- a/test/replay_tests/basic_replay_multiple_fixtures_and_params.py
+++ b/test/replay_tests/basic_replay_multiple_fixtures_and_params.py
@@ -29,10 +29,10 @@ from replay_testing import (
     run,
 )
 
-replay_testing_dir = get_package_share_directory("replay_testing")
+replay_testing_dir = get_package_share_directory('replay_testing')
 
-cmd_vel_only_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only.mcap")
-cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only_2.mcap")
+cmd_vel_only_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only.mcap')
+cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only_2.mcap')
 
 
 @fixtures.parameterize([
@@ -40,35 +40,35 @@ cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "c
     McapFixture(path=cmd_vel_only_2_fixture),
 ])
 class Fixtures:
-    required_input_topics = ["/vehicle/cmd_vel"]
-    expected_output_topics = ["/user/cmd_vel"]
+    required_input_topics = ['/vehicle/cmd_vel']
+    expected_output_topics = ['/user/cmd_vel']
 
 
 @run.parameterize([
-    ReplayRunParams(name="run_1_twist_slow", params={"x": 1.0}),
-    ReplayRunParams(name="run_2_twist_fast", params={"x": 10.0}),
+    ReplayRunParams(name='run_1_twist_slow', params={'x': 1.0}),
+    ReplayRunParams(name='run_2_twist_fast', params={'x': 10.0}),
 ])
 class Run:
     def generate_launch_description(self, replay_run_params: ReplayRunParams) -> LaunchDescription:
-        print("replay_run_parms")
+        print('replay_run_parms')
         twist_msg = {
-            "linear": {"x": replay_run_params.params["x"]},
-            "angular": {"z": 0.5},
+            'linear': {'x': replay_run_params.params['x']},
+            'angular': {'z': 0.5},
         }
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    "ros2",
-                    "topic",
-                    "pub",
-                    "-r",
-                    "10",
-                    "/user/cmd_vel",
-                    "geometry_msgs/msg/Twist",
+                    'ros2',
+                    'topic',
+                    'pub',
+                    '-r',
+                    '10',
+                    '/user/cmd_vel',
+                    'geometry_msgs/msg/Twist',
                     json.dumps(twist_msg),
                 ],
-                name="topic_pub",
-                output="screen",
+                name='topic_pub',
+                output='screen',
             )
         ])
 
@@ -76,8 +76,8 @@ class Run:
 @analyze
 class Analyze:
     def test_cmd_vel(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
         msgs = [msg for msg in msgs_it]
         assert len(msgs) >= 1
-        assert msgs[0].channel.topic == "/user/cmd_vel"
+        assert msgs[0].channel.topic == '/user/cmd_vel'

--- a/test/replay_tests/nexus_replay.py
+++ b/test/replay_tests/nexus_replay.py
@@ -20,10 +20,10 @@ from launch.actions import ExecuteProcess
 from replay_testing import NexusFixture, analyze, fixtures, run
 
 
-@fixtures.parameterize([NexusFixture(path='generic/cmd_vel_only.mcap')])
+@fixtures.parameterize([NexusFixture(path="generic/cmd_vel_only.mcap")])
 class Fixtures:
-    required_input_topics = ['/vehicle/cmd_vel']
-    expected_output_topics = ['/user/cmd_vel']
+    required_input_topics = ["/vehicle/cmd_vel"]
+    expected_output_topics = ["/user/cmd_vel"]
 
 
 @run.default()
@@ -32,15 +32,17 @@ class Run:
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    'ros2',
-                    'topic',
-                    'pub',
-                    '/user/cmd_vel',
-                    'geometry_msgs/msg/Twist',
-                    '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                    "ros2",
+                    "topic",
+                    "pub",
+                    "-r",
+                    "10",
+                    "/user/cmd_vel",
+                    "geometry_msgs/msg/Twist",
+                    "{linear: {x: 1.0}, angular: {z: 0.5}}",
                 ],
-                name='topic_pub',
-                output='screen',
+                name="topic_pub",
+                output="screen",
             )
         ])
 
@@ -48,8 +50,8 @@ class Run:
 @analyze
 class AnalyzeBasicReplay:
     def test_cmd_vel(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
         msgs = [msg for msg in msgs_it]
-        assert len(msgs) == 1
-        assert msgs[0].channel.topic == '/user/cmd_vel'
+        assert len(msgs) >= 1
+        assert msgs[0].channel.topic == "/user/cmd_vel"

--- a/test/replay_tests/nexus_replay.py
+++ b/test/replay_tests/nexus_replay.py
@@ -20,10 +20,10 @@ from launch.actions import ExecuteProcess
 from replay_testing import NexusFixture, analyze, fixtures, run
 
 
-@fixtures.parameterize([NexusFixture(path="generic/cmd_vel_only.mcap")])
+@fixtures.parameterize([NexusFixture(path='generic/cmd_vel_only.mcap')])
 class Fixtures:
-    required_input_topics = ["/vehicle/cmd_vel"]
-    expected_output_topics = ["/user/cmd_vel"]
+    required_input_topics = ['/vehicle/cmd_vel']
+    expected_output_topics = ['/user/cmd_vel']
 
 
 @run.default()
@@ -32,17 +32,17 @@ class Run:
         return LaunchDescription([
             ExecuteProcess(
                 cmd=[
-                    "ros2",
-                    "topic",
-                    "pub",
-                    "-r",
-                    "10",
-                    "/user/cmd_vel",
-                    "geometry_msgs/msg/Twist",
-                    "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                    'ros2',
+                    'topic',
+                    'pub',
+                    '-r',
+                    '10',
+                    '/user/cmd_vel',
+                    'geometry_msgs/msg/Twist',
+                    '{linear: {x: 1.0}, angular: {z: 0.5}}',
                 ],
-                name="topic_pub",
-                output="screen",
+                name='topic_pub',
+                output='screen',
             )
         ])
 
@@ -50,8 +50,8 @@ class Run:
 @analyze
 class AnalyzeBasicReplay:
     def test_cmd_vel(self):
-        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+        msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
         msgs = [msg for msg in msgs_it]
         assert len(msgs) >= 1
-        assert msgs[0].channel.topic == "/user/cmd_vel"
+        assert msgs[0].channel.topic == '/user/cmd_vel'

--- a/test/test_replay_runner.py
+++ b/test/test_replay_runner.py
@@ -34,18 +34,18 @@ from replay_testing import (
     run,
 )
 
-replay_testing_dir = get_package_share_directory("replay_testing")
+replay_testing_dir = get_package_share_directory('replay_testing')
 
-cmd_vel_only_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only.mcap")
-cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only_2.mcap")
+cmd_vel_only_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only.mcap')
+cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only_2.mcap')
 
 
 def test_fixtures():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel"]
+        required_input_topics = ['/vehicle/cmd_vel']
         expected_output_topics = []
 
     test_module.Fixtures = Fixtures
@@ -60,16 +60,16 @@ def test_fixtures():
     topics = reader.get_all_topics_and_types()
 
     assert len(topics) == 1
-    assert topics[0].name == "/vehicle/cmd_vel"
+    assert topics[0].name == '/vehicle/cmd_vel'
     return
 
 
 def test_fixtures_raises_err():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel", "/does_not_exist"]
+        required_input_topics = ['/vehicle/cmd_vel', '/does_not_exist']
         expected_output_topics = []
 
     test_module.Fixtures = Fixtures
@@ -80,12 +80,12 @@ def test_fixtures_raises_err():
 
 
 def test_run():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel"]
-        expected_output_topics = ["/user/cmd_vel"]
+        required_input_topics = ['/vehicle/cmd_vel']
+        expected_output_topics = ['/user/cmd_vel']
 
     @run.default()
     class Run:
@@ -93,17 +93,17 @@ def test_run():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        "ros2",
-                        "topic",
-                        "pub",
-                        "-r",
-                        "10",
-                        "/user/cmd_vel",
-                        "geometry_msgs/msg/Twist",
-                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                        'ros2',
+                        'topic',
+                        'pub',
+                        '-r',
+                        '10',
+                        '/user/cmd_vel',
+                        'geometry_msgs/msg/Twist',
+                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
                     ],
-                    name="topic_pub",
-                    output="screen",
+                    name='topic_pub',
+                    output='screen',
                 )
             ])
 
@@ -121,25 +121,25 @@ def test_run():
     topics = reader.get_all_topics_and_types()
     topic_names = [topic.name for topic in topics]
 
-    assert "/vehicle/cmd_vel" in topic_names
-    assert "/user/cmd_vel" in topic_names
+    assert '/vehicle/cmd_vel' in topic_names
+    assert '/user/cmd_vel' in topic_names
 
     msg_reader = get_message_mcap_reader(run_fixture.path)
-    msgs_it = mcap_ros2.reader.read_ros2_messages(msg_reader, topics=["/user/cmd_vel"])
+    msgs_it = mcap_ros2.reader.read_ros2_messages(msg_reader, topics=['/user/cmd_vel'])
 
     msgs = [msg for msg in msgs_it]
     assert len(msgs) >= 1
-    assert msgs[0].channel.topic == "/user/cmd_vel"
+    assert msgs[0].channel.topic == '/user/cmd_vel'
     return
 
 
 def test_analyze():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel"]
-        expected_output_topics = ["/user/cmd_vel"]
+        required_input_topics = ['/vehicle/cmd_vel']
+        expected_output_topics = ['/user/cmd_vel']
 
     @run.default()
     class Run:
@@ -147,28 +147,28 @@ def test_analyze():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        "ros2",
-                        "topic",
-                        "pub",
-                        "-r",
-                        "10",
-                        "/user/cmd_vel",
-                        "geometry_msgs/msg/Twist",
-                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                        'ros2',
+                        'topic',
+                        'pub',
+                        '-r',
+                        '10',
+                        '/user/cmd_vel',
+                        'geometry_msgs/msg/Twist',
+                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
                     ],
-                    name="topic_pub",
-                    output="screen",
+                    name='topic_pub',
+                    output='screen',
                 )
             ])
 
     @analyze
     class Analyze:
         def test_cmd_vel(self):
-            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
             msgs = [msg for msg in msgs_it]
             assert len(msgs) >= 1
-            assert msgs[0].channel.topic == "/user/cmd_vel"
+            assert msgs[0].channel.topic == '/user/cmd_vel'
 
     test_module.Fixtures = Fixtures
     test_module.Run = Run
@@ -184,12 +184,12 @@ def test_analyze():
 
 
 def test_failed_analyze():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel"]
-        expected_output_topics = ["/user/cmd_vel"]
+        required_input_topics = ['/vehicle/cmd_vel']
+        expected_output_topics = ['/user/cmd_vel']
 
     @run.default()
     class Run:
@@ -197,17 +197,17 @@ def test_failed_analyze():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        "ros2",
-                        "topic",
-                        "pub",
-                        "-r",
-                        "10",
-                        "/user/cmd_vel",
-                        "geometry_msgs/msg/Twist",
-                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                        'ros2',
+                        'topic',
+                        'pub',
+                        '-r',
+                        '10',
+                        '/user/cmd_vel',
+                        'geometry_msgs/msg/Twist',
+                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
                     ],
-                    name="topic_pub",
-                    output="screen",
+                    name='topic_pub',
+                    output='screen',
                 )
             ])
 
@@ -230,15 +230,15 @@ def test_failed_analyze():
 
 
 def test_multiple_fixtures():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([
         McapFixture(path=cmd_vel_only_fixture),
         McapFixture(path=cmd_vel_only_2_fixture),
     ])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel"]
-        expected_output_topics = ["/user/cmd_vel"]
+        required_input_topics = ['/vehicle/cmd_vel']
+        expected_output_topics = ['/user/cmd_vel']
 
     @run.default()
     class Run:
@@ -246,28 +246,28 @@ def test_multiple_fixtures():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        "ros2",
-                        "topic",
-                        "pub",
-                        "-r",
-                        "10",
-                        "/user/cmd_vel",
-                        "geometry_msgs/msg/Twist",
-                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
+                        'ros2',
+                        'topic',
+                        'pub',
+                        '-r',
+                        '10',
+                        '/user/cmd_vel',
+                        'geometry_msgs/msg/Twist',
+                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
                     ],
-                    name="topic_pub",
-                    output="screen",
+                    name='topic_pub',
+                    output='screen',
                 )
             ])
 
     @analyze
     class Analyze:
         def test_cmd_vel(self):
-            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
             msgs = [msg for msg in msgs_it]
             assert len(msgs) >= 1
-            assert msgs[0].channel.topic == "/user/cmd_vel"
+            assert msgs[0].channel.topic == '/user/cmd_vel'
 
     test_module.Fixtures = Fixtures
     test_module.Run = Run
@@ -291,51 +291,49 @@ def test_multiple_fixtures():
 
 
 def test_parametric_sweep():
-    test_module = types.ModuleType("test_module")
+    test_module = types.ModuleType('test_module')
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ["/vehicle/cmd_vel"]
-        expected_output_topics = ["/user/cmd_vel"]
+        required_input_topics = ['/vehicle/cmd_vel']
+        expected_output_topics = ['/user/cmd_vel']
 
     @run.parameterize([
-        ReplayRunParams(name="run_1_twist_slow", params={"x": 1.0}),
-        ReplayRunParams(name="run_2_twist_fast", params={"x": 10.0}),
+        ReplayRunParams(name='run_1_twist_slow', params={'x': 1.0}),
+        ReplayRunParams(name='run_2_twist_fast', params={'x': 10.0}),
     ])
     class Run:
-        def generate_launch_description(
-            self, replay_run_params: ReplayRunParams
-        ) -> LaunchDescription:
-            print("replay_run_parms")
+        def generate_launch_description(self, replay_run_params: ReplayRunParams) -> LaunchDescription:
+            print('replay_run_parms')
             twist_msg = {
-                "linear": {"x": replay_run_params.params["x"]},
-                "angular": {"z": 0.5},
+                'linear': {'x': replay_run_params.params['x']},
+                'angular': {'z': 0.5},
             }
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        "ros2",
-                        "topic",
-                        "pub",
-                        "-r",
-                        "10",
-                        "/user/cmd_vel",
-                        "geometry_msgs/msg/Twist",
+                        'ros2',
+                        'topic',
+                        'pub',
+                        '-r',
+                        '10',
+                        '/user/cmd_vel',
+                        'geometry_msgs/msg/Twist',
                         json.dumps(twist_msg),
                     ],
-                    name="topic_pub",
-                    output="screen",
+                    name='topic_pub',
+                    output='screen',
                 )
             ])
 
     @analyze
     class Analyze:
         def test_cmd_vel(self):
-            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
+            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
 
             msgs = [msg for msg in msgs_it]
             assert len(msgs) >= 1
-            assert msgs[0].channel.topic == "/user/cmd_vel"
+            assert msgs[0].channel.topic == '/user/cmd_vel'
 
     test_module.Fixtures = Fixtures
     test_module.Run = Run

--- a/test/test_replay_runner.py
+++ b/test/test_replay_runner.py
@@ -34,18 +34,18 @@ from replay_testing import (
     run,
 )
 
-replay_testing_dir = get_package_share_directory('replay_testing')
+replay_testing_dir = get_package_share_directory("replay_testing")
 
-cmd_vel_only_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only.mcap')
-cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, 'test', 'fixtures', 'cmd_vel_only_2.mcap')
+cmd_vel_only_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only.mcap")
+cmd_vel_only_2_fixture = os.path.join(replay_testing_dir, "test", "fixtures", "cmd_vel_only_2.mcap")
 
 
 def test_fixtures():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel']
+        required_input_topics = ["/vehicle/cmd_vel"]
         expected_output_topics = []
 
     test_module.Fixtures = Fixtures
@@ -60,16 +60,16 @@ def test_fixtures():
     topics = reader.get_all_topics_and_types()
 
     assert len(topics) == 1
-    assert topics[0].name == '/vehicle/cmd_vel'
+    assert topics[0].name == "/vehicle/cmd_vel"
     return
 
 
 def test_fixtures_raises_err():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel', '/does_not_exist']
+        required_input_topics = ["/vehicle/cmd_vel", "/does_not_exist"]
         expected_output_topics = []
 
     test_module.Fixtures = Fixtures
@@ -80,12 +80,12 @@ def test_fixtures_raises_err():
 
 
 def test_run():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel']
-        expected_output_topics = ['/user/cmd_vel']
+        required_input_topics = ["/vehicle/cmd_vel"]
+        expected_output_topics = ["/user/cmd_vel"]
 
     @run.default()
     class Run:
@@ -93,15 +93,17 @@ def test_run():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        'ros2',
-                        'topic',
-                        'pub',
-                        '/user/cmd_vel',
-                        'geometry_msgs/msg/Twist',
-                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                        "ros2",
+                        "topic",
+                        "pub",
+                        "-r",
+                        "10",
+                        "/user/cmd_vel",
+                        "geometry_msgs/msg/Twist",
+                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
                     ],
-                    name='topic_pub',
-                    output='screen',
+                    name="topic_pub",
+                    output="screen",
                 )
             ])
 
@@ -119,25 +121,25 @@ def test_run():
     topics = reader.get_all_topics_and_types()
     topic_names = [topic.name for topic in topics]
 
-    assert '/vehicle/cmd_vel' in topic_names
-    assert '/user/cmd_vel' in topic_names
+    assert "/vehicle/cmd_vel" in topic_names
+    assert "/user/cmd_vel" in topic_names
 
     msg_reader = get_message_mcap_reader(run_fixture.path)
-    msgs_it = mcap_ros2.reader.read_ros2_messages(msg_reader, topics=['/user/cmd_vel'])
+    msgs_it = mcap_ros2.reader.read_ros2_messages(msg_reader, topics=["/user/cmd_vel"])
 
     msgs = [msg for msg in msgs_it]
-    assert len(msgs) == 1
-    assert msgs[0].channel.topic == '/user/cmd_vel'
+    assert len(msgs) >= 1
+    assert msgs[0].channel.topic == "/user/cmd_vel"
     return
 
 
 def test_analyze():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel']
-        expected_output_topics = ['/user/cmd_vel']
+        required_input_topics = ["/vehicle/cmd_vel"]
+        expected_output_topics = ["/user/cmd_vel"]
 
     @run.default()
     class Run:
@@ -145,26 +147,28 @@ def test_analyze():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        'ros2',
-                        'topic',
-                        'pub',
-                        '/user/cmd_vel',
-                        'geometry_msgs/msg/Twist',
-                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                        "ros2",
+                        "topic",
+                        "pub",
+                        "-r",
+                        "10",
+                        "/user/cmd_vel",
+                        "geometry_msgs/msg/Twist",
+                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
                     ],
-                    name='topic_pub',
-                    output='screen',
+                    name="topic_pub",
+                    output="screen",
                 )
             ])
 
     @analyze
     class Analyze:
         def test_cmd_vel(self):
-            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
             msgs = [msg for msg in msgs_it]
-            assert len(msgs) == 1
-            assert msgs[0].channel.topic == '/user/cmd_vel'
+            assert len(msgs) >= 1
+            assert msgs[0].channel.topic == "/user/cmd_vel"
 
     test_module.Fixtures = Fixtures
     test_module.Run = Run
@@ -180,12 +184,12 @@ def test_analyze():
 
 
 def test_failed_analyze():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel']
-        expected_output_topics = ['/user/cmd_vel']
+        required_input_topics = ["/vehicle/cmd_vel"]
+        expected_output_topics = ["/user/cmd_vel"]
 
     @run.default()
     class Run:
@@ -193,15 +197,17 @@ def test_failed_analyze():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        'ros2',
-                        'topic',
-                        'pub',
-                        '/user/cmd_vel',
-                        'geometry_msgs/msg/Twist',
-                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                        "ros2",
+                        "topic",
+                        "pub",
+                        "-r",
+                        "10",
+                        "/user/cmd_vel",
+                        "geometry_msgs/msg/Twist",
+                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
                     ],
-                    name='topic_pub',
-                    output='screen',
+                    name="topic_pub",
+                    output="screen",
                 )
             ])
 
@@ -224,15 +230,15 @@ def test_failed_analyze():
 
 
 def test_multiple_fixtures():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([
         McapFixture(path=cmd_vel_only_fixture),
         McapFixture(path=cmd_vel_only_2_fixture),
     ])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel']
-        expected_output_topics = ['/user/cmd_vel']
+        required_input_topics = ["/vehicle/cmd_vel"]
+        expected_output_topics = ["/user/cmd_vel"]
 
     @run.default()
     class Run:
@@ -240,26 +246,28 @@ def test_multiple_fixtures():
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        'ros2',
-                        'topic',
-                        'pub',
-                        '/user/cmd_vel',
-                        'geometry_msgs/msg/Twist',
-                        '{linear: {x: 1.0}, angular: {z: 0.5}}',
+                        "ros2",
+                        "topic",
+                        "pub",
+                        "-r",
+                        "10",
+                        "/user/cmd_vel",
+                        "geometry_msgs/msg/Twist",
+                        "{linear: {x: 1.0}, angular: {z: 0.5}}",
                     ],
-                    name='topic_pub',
-                    output='screen',
+                    name="topic_pub",
+                    output="screen",
                 )
             ])
 
     @analyze
     class Analyze:
         def test_cmd_vel(self):
-            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
             msgs = [msg for msg in msgs_it]
-            assert len(msgs) == 1
-            assert msgs[0].channel.topic == '/user/cmd_vel'
+            assert len(msgs) >= 1
+            assert msgs[0].channel.topic == "/user/cmd_vel"
 
     test_module.Fixtures = Fixtures
     test_module.Run = Run
@@ -283,47 +291,51 @@ def test_multiple_fixtures():
 
 
 def test_parametric_sweep():
-    test_module = types.ModuleType('test_module')
+    test_module = types.ModuleType("test_module")
 
     @fixtures.parameterize([McapFixture(path=cmd_vel_only_fixture)])
     class Fixtures:
-        required_input_topics = ['/vehicle/cmd_vel']
-        expected_output_topics = ['/user/cmd_vel']
+        required_input_topics = ["/vehicle/cmd_vel"]
+        expected_output_topics = ["/user/cmd_vel"]
 
     @run.parameterize([
-        ReplayRunParams(name='run_1_twist_slow', params={'x': 1.0}),
-        ReplayRunParams(name='run_2_twist_fast', params={'x': 10.0}),
+        ReplayRunParams(name="run_1_twist_slow", params={"x": 1.0}),
+        ReplayRunParams(name="run_2_twist_fast", params={"x": 10.0}),
     ])
     class Run:
-        def generate_launch_description(self, replay_run_params: ReplayRunParams) -> LaunchDescription:
-            print('replay_run_parms')
+        def generate_launch_description(
+            self, replay_run_params: ReplayRunParams
+        ) -> LaunchDescription:
+            print("replay_run_parms")
             twist_msg = {
-                'linear': {'x': replay_run_params.params['x']},
-                'angular': {'z': 0.5},
+                "linear": {"x": replay_run_params.params["x"]},
+                "angular": {"z": 0.5},
             }
             return LaunchDescription([
                 ExecuteProcess(
                     cmd=[
-                        'ros2',
-                        'topic',
-                        'pub',
-                        '/user/cmd_vel',
-                        'geometry_msgs/msg/Twist',
+                        "ros2",
+                        "topic",
+                        "pub",
+                        "-r",
+                        "10",
+                        "/user/cmd_vel",
+                        "geometry_msgs/msg/Twist",
                         json.dumps(twist_msg),
                     ],
-                    name='topic_pub',
-                    output='screen',
+                    name="topic_pub",
+                    output="screen",
                 )
             ])
 
     @analyze
     class Analyze:
         def test_cmd_vel(self):
-            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=['/user/cmd_vel'])
+            msgs_it = mcap_ros2.reader.read_ros2_messages(self.reader, topics=["/user/cmd_vel"])
 
             msgs = [msg for msg in msgs_it]
-            assert len(msgs) == 1
-            assert msgs[0].channel.topic == '/user/cmd_vel'
+            assert len(msgs) >= 1
+            assert msgs[0].channel.topic == "/user/cmd_vel"
 
     test_module.Fixtures = Fixtures
     test_module.Run = Run

